libraries/chain/backing_store/tests/benchmark_kv.cpp

```diff
@@ -0,0 +1,337 @@
+   create_key_values(args, kv_context_ptr);
+   std::vector<std::string> work_set = get_work_set_keys(args);
+
+   if (args.operation == "get" ) {
+      benchmark_get(args, std::move(kv_context_ptr), work_set, start, end, usage_start, usage_end);
+   } else if (args.operation == "get_data" ) {
+      benchmark_get_data(args, std::move(kv_context_ptr), work_set, start, end, usage_start, usage_end);
+   } else if (args.operation == "set" ) {
+      benchmark_set(args, std::move(kv_context_ptr), work_set, start, end, usage_start, usage_end);
+   } else if (args.operation == "it_next" ) {
+      benchmark_it_next(args, std::move(kv_context_ptr), start, end, usage_start, usage_end);
+   } else if (args.operation == "it_key_value" ) {
+      benchmark_it_key_value(args, std::move(kv_context_ptr), start, end, usage_start, usage_end);
+   };
+}
+
+void print_results(cmd_args& args, wall_clock_t& start, wall_clock_t& end, const rusage& usage_start, const rusage& usage_end) {
+   std::string title = "Benchmarked \"" + args.operation + "\" for ";
+
+   if (args.operation == "it_next" || args.operation == "it_key_value") {
+      title += " one time over ";
+   } else {
+      title += std::to_string(num_iterations);
+      title += " times on ";
+      title += std::to_string(work_set_size);
+      title += " keys from ";
+   }
+
+   title += std::to_string(args.num_keys);
+   title += " total keys ";
+
+   if (args.is_rocksdb) {
+      title += "on RocksDb:";
+   } else {
+      title += "on Chainbase:";
+   }
+
+   uint32_t clock_duration_us = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+   uint32_t user_duration_us = 1000000 * (usage_end.ru_utime.tv_sec - usage_start.ru_utime.tv_sec) + usage_end.ru_utime.tv_usec - usage_start.ru_utime.tv_usec;;
+   uint32_t system_duration_us = 1000000 * (usage_end.ru_stime.tv_sec - usage_start.ru_stime.tv_sec) + usage_end.ru_stime.tv_usec - usage_start.ru_stime.tv_usec;;
+   uint32_t minor_faults = uint32_t(usage_end.ru_minflt  - usage_start.ru_minflt);
+   uint32_t major_faults = uint32_t(usage_end.ru_majflt  - usage_start.ru_majflt);
+   uint32_t blocks_in = uint32_t(usage_end.ru_inblock - usage_start.ru_inblock);
+   uint32_t blocks_out = uint32_t(usage_end.ru_oublock - usage_start.ru_oublock);
+
+   std::cout << title << std::endl 
+      << "   clock_duration_us: " << clock_duration_us
+      << ", user_duration_us: " << user_duration_us 
+      << ", system_duration_us: " << system_duration_us 
+      << ", minor_faults: " << minor_faults 
+      << ", major_faults: " << major_faults 
+      << ", blocks_in: " << blocks_in  
+      << ", blocks_out: " << blocks_out 
+      << std::endl << std::endl;
+}
+
+void benchmark(cmd_args& args) {
+   uint32_t clock_duration_us;
+   wall_clock_t start, end;
+   struct rusage usage_start, usage_end;
+
+   if (args.is_rocksdb) {
+      boost::filesystem::remove_all("kvrdb-tmp");  // Use a clean RocksDB
```

Comment:
rocksdb has a method for deleting the database files given the database name.  I don't know if there are other files that need to be removed.

Reply:
Will look around for how to start with a clean database.